src/main/java/org/truffleruby/core/kernel/KernelNodes.java

```diff
@@ -664,6 +657,32 @@ protected boolean assignsNoNewVariables(RootNodeWrapper rootNode) {
         private boolean frameHasOnlySelf(final FrameDescriptor descriptor) {
             return descriptor.getSize() == 1 && SelfNode.SELF_IDENTIFIER.equals(descriptor.getSlots().get(0).getIdentifier());
         }
+
+        @TruffleBoundary
+        public static String offsetSource(String method, String source, String file, int line) {
+            // TODO CS 23-Apr-18 Truffle doesn't support line numbers starting at anything but 1
+            if (line == 0) {
+                // fine instead of warning because these seem common
+                Log.LOGGER.fine(() -> String.format("zero line number %s:%d not supported in #%s - will be reported as starting at 1", file, line, method));
+                return source;
+            } else if (line < 1) {
+                Log.LOGGER.warning(String.format("negative line number %s:%d not supported in #%s - will be reported as starting at 1", file, line, method));
+                return source;
+            } else if (line > 1) {
+                // fine instead of warning because we can simulate these
+                Log.LOGGER.fine(() -> String.format("offset line number %s:%d are simulated in #%s by adding blank lines", file, line, method));
```

Comment:
Why warning here? Does it have any user-visible side effect?
If not, I think it's just our internal, but the user does not need to know about it and so we shouldn't warn.

Reply:
It'll show up in things like tooling, where you may get screens of empty text. I could also write a `TODO` comment so we remember something weird is going on but I'm trying to move away from doing that as we start to think about releases, so that weird things show up in users' logs and we can find them.

Reply:
I see, makes sense.
Maybe we should have different levels to differentiate higher-level initialization messages (like initalizeContext()).

Reply:
Different logging levels? This is already `fine` so it doesn't show up by default.